v3/src/components/web-view/web-view-utils.ts

```diff
@@ -0,0 +1,23 @@
+export const kRootPluginUrl = "https://codap-resources.s3.amazonaws.com/plugins"
+export const kRelativePluginRoot = "../../../../extn/plugins"
+
+export function processPluginUrl(url: string) {
+  let updatedUrl = url
+
+  // Some plugins relied on index.html being the default file loaded when pointing to a directory.
+  // This is no longer the case with our S3 hosted plugins, so we have to modify the path to point to the
+  // correct file in this case.
+  if (updatedUrl.endsWith("/")) updatedUrl = `${updatedUrl}index.html`
+
+  // Many plugins used to be packaged with CODAP v2, then referenced relatively.
+  // These references need to be changed to point to the S3 bucket we now use to host plugins.
+  if (updatedUrl.startsWith(kRelativePluginRoot)) {
+    updatedUrl = `${kRootPluginUrl}${updatedUrl.slice(kRelativePluginRoot.length)}`
+  }
+
+  // Some plugins were referenced with "http://", but this is no longer secure enough in the world of https.
+  const http = "http://"
+  if (updatedUrl.startsWith(http)) updatedUrl = `https://${updatedUrl.slice(http.length)}`
```

Comment:
I'm certainly fine with this for any plugin urls that we control. Given that plugins can be developed/served by anyone, there's a potential compatibility issue here for outside plugins that could, in theory, be served by servers that don't support `https`. In short, I think this is fine for now, but it may not be the last word on the subject.